### DISCUSSION TITLE
X-Callback-URL question

### DISCUSSION THREAD

**Discussion by Eidansoft** [BLeeEZ/amperfy#163] [Q&A] [answered]
Hi, 
I'm using the X-Callback-URL to play music into Amperfy from the IPhone shortcuts, but I can't just play random songs from my library with the callback, because the `SearchAndPlay` method has mandatory the `searchTerm`. But I don't want to serach by any term, I just want play randomly.

This command works but only returns songs that matches with `Example`:
`amperfy://x-callback-url/searchAndPlay?searchTerm=Example&searchCategory=song&shuffleOption=1`

If I try removing the `searchTerm` it will fails, saying the `searchTerm` is mandatory. So below do not work.
`amperfy://x-callback-url/searchAndPlay?searchCategory=song&shuffleOption=1`

I also have tried the `play` command, but it just does play the current playlist that do not necessarily is randomized or it was just a single song playlist.
`amperfy://x-callback-url/play`

So my question is how can I just trigger to play random songs with a callback, without need to provide any searchTerm??, just like you can do from `Library` -> `Songs` -> `Play random songs` button at the UI interface.

**Comment by BLeeEZ** [comment, marked as answer]
Hi, play random songs is currently not provided as x-callback-url. This is a great feature I will add in the next version, maybe I even add it to the next beta version.

**Reply by Eidansoft** [reply]
Great to know! Thanks!

**Comment by BLeeEZ** [comment]
This feature is now implemented in 66178322e1079f61508b047b35d015202ceadb8d:

"amperfy://x-callback-url/playRandomSongs?onlyCached=1"
onlyCached: optional -> "0 (false) or 1 (true), use only cached songs from library",

**Reply by Eidansoft** [reply]
Awesome!

Thank you @BLeeEZ , any ETA when that change will be included in the AppStore?

I’m thinking I should learn swift just to be able to collaborate with Amperfy! It’s embarrassing be a developer but not able to add/modify code to my own mobile.
